### DISCUSSION TITLE
Opt-out of pages router

### DISCUSSION THREAD

**Discussion by wouterraateland** [vercel/next.js#52946] [App Router] [answered]
When running `next dev` or `next build`, the pages router is always started even if only the app router is used.

Would be nice to be able to opt-out of the pages router, i.e.:

```JavaScript
{ experimental: { pagesDir: false } }
```

Potential benefits:
- Saves memory during dev: 1 process less
- Saves build time
- Might be easier on the router process during dev

**Comment by icyJoseph** [comment, marked as answer]
I do think they are, slowly but surely, moving in the direction of not doing pages work, or some of it anyway, if we use only app router:

- https://github.com/vercel/next.js/issues/49796#issuecomment-1547628161

Though since they are supporting a "migration" phase between both routers, I could imagine, it'll take some time for them to be able to decouple a bunch of code. I mean, there's some areas where they do `if (appRouter)` checks, but perhaps in the future, they can assume to be in appRoute instead, and check for pages router.

**Reply by wouterraateland** [reply]
Makes sense! Couldn't find it in the issues yet, so thought I'd suggest. Thanks for digging that up.
